sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -244,7 +243,10 @@ object DataSourceAnalysis extends Rule[LogicalPlan] {
 class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan] {
   private def readDataSourceTable(
       table: CatalogTable, extraOptions: CaseInsensitiveStringMap): LogicalPlan = {
-    val qualifiedTableName = QualifiedTableName(table.database, table.identifier.table)
+    val catalogName =
+      format(table.identifier.catalog.getOrElse(CatalogManager.SESSION_CATALOG_NAME))
```

Comment:
The identifier must be qualified when we reach here. We can just do `table.identifier.catalog.get`. We did the same thing for `database`

Reply:
ack